bpf/lib/common.h

```diff
@@ -672,6 +676,20 @@ enum metric_dir {
 #define DSR_IPV6_OPT_LEN	(sizeof(struct dsr_opt_v6) - 4)
 #define DSR_IPV6_EXT_LEN	((sizeof(struct dsr_opt_v6) - 8) / 8)
 
+/* Geneve option used to carry service addr and port for DSR.
+ *
+ * Class = 0x200 (unassigned according to [1])
+ * Type  = 0x1   (vendor-specific)
+ *
+ * [1]: https://www.iana.org/assignments/nvo3/nvo3.xhtml#geneve-option-class
+ */
+#define DSR_GENEVE_OPT_CLASS	0x200
+#define DSR_GENEVE_OPT_TYPE	0x1
```

Comment:
Were you able to use wireshark or something to confirm whether this is the "Critical bit" with opt type 0, or whether this is opt type 1 with no critical bit?

Reply:
It's opt type 1 with no critical bit. I will add critical bit next

Reply:
I need to set `bpf_tunnel_key.tunnel_flags`, perhaps
https://github.com/torvalds/linux/blob/1e760fa3596e8c7f08412712c168288b79670d78/drivers/net/geneve.c#L747

Reply:
As far as I looked into this, we can't set `TUNNEL_CRIT_OPT` to `ip_tunnel_info.key.tun_flags`
https://github.com/torvalds/linux/blob/1e760fa3596e8c7f08412712c168288b79670d78/net/core/filter.c#L4643-L4712

According to the Open vSwitch implementation, it sets  `TUNNEL_CRIT_OPT` to `ip_tunnel_info.key.tun_flags` if one or more options have the critical bit set
https://github.com/torvalds/linux/blob/1e760fa3596e8c7f08412712c168288b79670d78/net/openvswitch/flow_netlink.c#L2722-L2748

But there's no corresponding implementation here
https://github.com/torvalds/linux/blob/1e760fa3596e8c7f08412712c168288b79670d78/drivers/net/geneve.c#L740-L755

IMHO, we can't support Critical Bit for now. What do you think?

Reply:
Critical Option present: False
0x81 = 10000001 (Critical Bit + type 1)

<img width="810" alt="geneve" src="https://user-images.githubusercontent.com/5572688/227766239-92b3d7e5-e443-4836-9806-5334ea6ba684.png">

Reply:
Ah I see. Adding to the option is fine for now. It looks like there's a "good first issue" for the kernel API here to either let programs provide the critical bit as a flag, or to scan the options and set the critical bit before tx. I would encourage you to follow up with the kernel folks on this if you can.

Reply:
>  I would encourage you to follow up with the kernel folks on this if you can.

Sure. I haven't sent a patch for the kernel API, but I will try.